### DISCUSSION TITLE
v19.5.0-rc4

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3265] [Releases]
## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Highlights
This is a big release, that's why we decided to move away from the 19.4.x versioning and go to 19.5.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 19.5.0. Tests are more than welcome now but be extra careful with production environment.

- What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.
- PHP 7.4 is now the minimum required version and 8.2 is now supported
- M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- Support for Google Analytics 4 was added

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

## Changelog
- Fixed session renew timestamp should be updated when customer changes password #2916
- Fixed price filter when search by non-numeric value #3136
- Added patch for Zend_Mail and "emails are displayed incorrectly" problem #3202

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v19.5.0-rc3...v19.5.0-rc4

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v19.5.0-rc4'>v19.5.0-rc4</a>.</em>

**Comment by djixas** [comment]
After upgrading from Magento 1.9, my front and backend gives error (even though i didn't even use redis nor does it have references in local.xml), upgrading from 1.9 to 19.40 works fine.

There has been an error processing your request
Unable to connect to Redis


```
Trace:
#0 /home/fffff/public_html/app/code/community/Cm/RedisSession/Model/Session.php(55): Cm\RedisSession\Handler->__construct(Object(Cm_RedisSession_Model_Session_Config), Object(Cm_RedisSession_Model_Session_Logger), false)
#1 /home/fffff/public_html/app/code/core/Mage/Core/Model/Config.php(1441): Cm_RedisSession_Model_Session->__construct(Array)
#2 /home/fffff/public_html/app/code/core/Mage/Core/Model/Config.php(1477): Mage_Core_Model_Config->getModelInstance('core_resource/s...', Array)
#3 /home/fffff/public_html/app/Mage.php(551): Mage_Core_Model_Config->getResourceModelInstance('core/session', Array)
#4 /home/fffff/public_html/app/Mage.php(579): Mage::getResourceModel('core/session', Array)
#5 /home/fffff/public_html/app/code/core/Mage/Core/Model/Session/Abstract/Varien.php(67): Mage::getResourceSingleton('core/session')
#6 /home/fffff/public_html/app/code/core/Mage/Core/Model/Session/Abstract/Varien.php(252): Mage_Core_Model_Session_Abstract_Varien->start('om_frontend')
#7 /home/fffff/public_html/app/code/core/Mage/Core/Model/Session/Abstract.php(80): Mage_Core_Model_Session_Abstract_Varien->init('core', 'om_frontend')
#8 /home/fffff/public_html/app/code/core/Mage/Core/Model/Session.php(44): Mage_Core_Model_Session_Abstract->init('core', 'om_frontend')
#9 /home/fffff/public_html/app/code/core/Mage/Core/Model/Config.php(1441): Mage_Core_Model_Session->__construct(Array)
#10 /home/fffff/public_html/app/Mage.php(523): Mage_Core_Model_Config->getModelInstance('core/session', Array)
#11 /home/fffff/public_html/app/Mage.php(537): Mage::getModel('core/session', Array)
#12 /home/fffff/public_html/app/code/core/Mage/Core/Controller/Varien/Action.php(496): Mage::getSingleton('core/session', Array)
#13 /home/fffff/public_html/app/code/core/Mage/Core/Controller/Front/Action.php(58): Mage_Core_Controller_Varien_Action->preDispatch()
#14 /home/fffff/public_html/app/code/core/Mage/Core/Controller/Varien/Action.php(411): Mage_Core_Controller_Front_Action->preDispatch()
#15 /home/fffff/public_html/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(256): Mage_Core_Controller_Varien_Action->dispatch('index')
#16 /home/fffff/public_html/app/code/core/Mage/Core/Controller/Varien/Front.php(182): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#17 /home/fffff/public_html/app/code/core/Mage/Core/Model/App.php(365): Mage_Core_Controller_Varien_Front->dispatch()
#18 /home/fffff/public_html/app/Mage.php(742): Mage_Core_Model_App->run(Array)
#19 /home/fffff/public_html/index.php(55): Mage::run('', 'store')
#20 {main}
```

**Reply by addison74** [reply]
You should check if the module exists in the file system. Cm_Redis no longer exists in /app/code/community. If you clone the repository you will have immediate problems.

**Reply by djixas** [reply]
I'm still unable to upgrade somehow even with official releases, i don't even use redis and i get this error on three different sites

EDIT: I think it's linked to https://github.com/colinmollenhour/Cm_Cache_Backend_Redis

If anyone ever has this issue

**Reply by luigifab** [reply]
If you don't use Redis, you can remove the module, or simply disable it.
